library/OpenTelemetry/AWSXRay/Baggage.hs

```diff
@@ -0,0 +1,52 @@
+-- | Functions for dealing with 'Baggage'
+--
+-- We ported specs about 'Baggage' from the Java Propagator:
+--
+-- <https://github.com/open-telemetry/opentelemetry-java-contrib/blob/04a2a481934be614e73e41194107ffdd767bc507/aws-xray-propagator/src/test/java/io/opentelemetry/contrib/awsxray/propagator/AwsXrayPropagatorTest.java#L227-L238>
+--
+-- And using the obvious 'encodeBaggageHeader'/'decodeBaggageHeader' functions
```

Comment:
That's too bad... Is it worth encapsulating these in our specs somewhere so they're not part of "the API we maintain" :tm: .

Reply:
I'm not sure what you mean, can you rephrase?

Reply:
So right now these would be exposed by our package but I read these comments as saying that they're for testing.

So I'm suggesting not making them part of our exposed interface since there's typically more potential maintenance burden/headache, the more you expose.

Reply:
>  I read these comments as saying that they're for testing.

Hmm, no, they're used by the real thing to de/encode 'Baggage'.

That said, they are _internal_. But I typically just don't try to slice up my package with public/private lines. I know some folks like to make a `.Internal`, or even maintain `modules` vs `extra-modules`, but I've never found that worthwhile. If I were going to go down that route, more than just these functions could be private. Only the `IdGenerator` and `Propogator` (and the `Setup` stuff) are actually necessary public API.

Reply:
Got it.